src/libraries/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/CryptoStream.cs

```diff
@@ -202,7 +203,7 @@ public override void SetLength(long value)
         public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
             CheckReadArguments(buffer, offset, count);
-            return ReadAsyncInternal(buffer, offset, count, cancellationToken);
+            return ReadAsync(buffer.AsMemory(offset, count), cancellationToken).AsTask();
```

Comment:
If we're concerned about that, both overloads of ReadAsync can delegate to a non-virtual ReadAsyncCore.

I've never seen a type derived from CryptoStream, though.  I'm sure someone somewhere has done it, but I don't think we need to be too concerned (famous last words).  If we were actually concerned, we'd potentially want to take it a step further and have the ReadAsync memory overload check whether the type was derived or not, delegating to the base implementation if it was, in case someone had overridden ReadAsync(byte[], ...) to do something special, in which case it would be a breaking change for ReadAsync(Memory, ...) to not use it.

Reply:
Would we want to do reflection to see if the derived type has actually overridden ReadAsync(byte[]...) so we can skip the memory copy if the derived type when possible?

Reply:
> Would we want to do reflection

No